### ISSUE TITLE
[FR] Make the experiment run Columns selector resizable

### ISSUE THREAD

**Issue by LockedThread** [mlflow/mlflow#26114]
> [!WARNING]
> Before submitting a PR, please make sure that:
> - A maintainer has triaged this issue and applied the `ready` label
> - This issue has no assignee
> - No duplicate PR exists
>
> PRs not meeting these requirements may be automatically closed.

### Willingness to contribute

Yes. I can contribute this feature independently.

### Proposal Summary

On the Model Training experiment page, the **Columns** selector currently has a fixed width of 400px, while the scrollable column tree is limited to a maximum height of 480px.

For experiments containing many metrics, parameters, and tags, this results in substantial scrolling. Long column names are also truncated, requiring users to hover over individual entries to determine their full names. This is particularly noticeable on large, ultrawide, and high-resolution displays where significantly more screen space is available but cannot be used by the selector.

I propose allowing users to resize the Columns selector horizontally and vertically on desktop.

Proposed behavior:

- Preserve approximately the current dimensions as the initial/default size.
- Allow the panel to be resized horizontally and vertically.
- Constrain resizing to the available viewport/popover area.
- Keep the column tree scrollable within the resized panel.
- Preserve the existing mobile/full-width behavior and disable manual resizing on small screens.
- Preserve existing search, keyboard navigation, column selection, and **Reset to defaults** behavior.

This would be a frontend-only enhancement to `ExperimentViewRunsColumnSelector` and would not require changes to experiment data or MLflow APIs.

A previous MLflow change, https://github.com/mlflow/mlflow/pull/2388, addressed aggressive truncation in this selector by increasing its width and improving hover behavior. Allowing user-controlled sizing would avoid requiring another globally fixed size while adapting better to different display sizes and experiment configurations.

I am willing to contribute the implementation.

#### What is the use case for this feature?

When analyzing experiments with a large number of parameters and metrics, users frequently need to enable and disable columns in the experiment runs table.

For larger experiments, the list in the **Columns** selector can contain dozens or hundreds of entries. Many parameter and metric names can also be relatively long, particularly when configuration values are represented using hierarchical names such as:

`synthetic_augmentation.generator.candidate_location...`

The current selector provides a relatively small fixed viewing area regardless of the available screen space. As a result, users must scroll through a small portion of the available column list and often hover over truncated entries to determine their complete names.

Allowing the panel to be resized would let users temporarily expand the selector while configuring their experiment table and then return to the normal interface afterward.

#### Why is this use case valuable to support for MLflow users in general?

MLflow is frequently used for hyperparameter optimization, large experiment comparisons, and workflows where a single run may contain many parameters, metrics, and tags.

As experiments become more complex, navigating experiment metadata becomes increasingly important. The existing search functionality helps when the user already knows the name of the column they are looking for, but it does not fully address workflows where users want to browse or compare the available metrics and parameters.

A resizable selector would:

- Make large experiment configurations easier to navigate.
- Reduce unnecessary scrolling.
- Improve readability of long parameter and metric names.
- Make better use of high-resolution and ultrawide displays.
- Preserve the compact default UI for users with smaller experiments.

This would improve the usability of the existing selector without changing its underlying behavior.

#### Why is this use case valuable to support for your project(s) or organization?

Our MLflow experiments can contain a large number of parameters and metrics, particularly when running hyperparameter optimization and comparing different model configurations.

Parameter names are often hierarchical and descriptive, which makes them considerably longer than the available width of the current selector. When reviewing a large experiment, I frequently need to inspect or enable several related metrics or parameters.

On an ultrawide/high-resolution display, there is substantial unused screen space available, but the selector remains limited to the same fixed dimensions. This makes browsing the available columns unnecessarily slow and often requires hovering over individual entries to identify them.

Being able to enlarge the selector would significantly improve this workflow while still allowing the existing compact size to remain the default.

#### Why is it currently difficult to achieve this use case?

The current `ExperimentViewRunsColumnSelector` uses a fixed panel width and a fixed maximum height for the scrollable tree.

Users cannot enlarge the selector even when additional viewport space is available.

The existing search field is useful when the exact parameter or metric name is already known, but it does not solve the broader browsing problem. Likewise, the tooltip shown for truncated names requires interacting with entries individually, which becomes cumbersome when reviewing many similarly named parameters or metrics.

There is currently no user-facing configuration or interaction that allows the selector to take advantage of additional available screen space.

### Details

A possible implementation would be limited primarily to:

`mlflow/server/js/src/experiment-tracking/components/experiment-page/components/runs/ExperimentViewRunsColumnSelector.tsx`

The current panel uses a fixed width:

```tsx
width: 400
```
and the scrollable column list uses a maximum height based on 15 tree elements:
```tsx
maxHeight: 15 * 32
```
One possible implementation would make the outer panel resizable on desktop while making the tree container flex to consume the available height.

For example, the outer container could use CSS resizing with appropriate minimum and maximum constraints:
```tsx
css={{
  width: 400,
  minWidth: 320,
  minHeight: 240,
  resize: 'both',
  overflow: 'hidden',
  display: 'flex',
  flexDirection: 'column',
}}
```
The column-list container could then become the flexible scrolling region:
```tsx
css={{
  flex: '1 1 auto',
  minHeight: 0,
  overflowY: 'auto',
  overflowX: 'hidden',
}}
```
The maximum dimensions should be constrained by the available viewport/popover area so that the panel cannot be resized off-screen.

For small-screen/mobile breakpoints, the existing full-width behavior could remain unchanged and manual resizing could be disabled.

Alternatively, if a native CSS resize control is not considered appropriate for the MLflow design system, the same behavior could be implemented using the resizing components already used elsewhere in the MLflow frontend.

I would initially avoid persisting the resized dimensions. The current size could remain the default whenever the selector is opened or the page is reloaded. Persisting the preferred dimensions could be considered separately if desired.

Existing behavior should remain unchanged for:

- Column search
- Selecting/deselecting columns
- Keyboard navigation
- Group selection
- Reset to defaults
- Popover positioning
- Small-screen/mobile layout

There is already targeted test coverage in:

https://github.com/mlflow/mlflow/blob/1563d1b8a00da7a6055854741d6c8e1bea6cc4f9/mlflow/server/js/src/experiment-tracking/components/experiment-page/components/runs/ExperimentViewRunsColumnSelector.test.tsx#L4

Any implementation could extend this coverage as appropriate and also include manual verification at normal desktop, ultrawide, and small-screen viewport sizes.

The attached screenshot demonstrates the current issue: a large amount of screen space is available while the Columns selector remains constrained to a relatively narrow and short viewing area.

<img width="434" height="597" alt="Image" src="https://github.com/user-attachments/assets/bd7e729e-dad9-4404-b0d9-670279c64f35" />

### What machine learning domain(s) is this feature request about?

- [ ] `domain/genai`: LLMs, Agents, and other GenAI-related use cases
- [x] `domain/classical-ml`: Traditional machine learning, such as linear regression.
- [x] `domain/deep-learning`: Deep learning and neural networks.
- [x] `domain/platform`: MLflow platform foundation, not specific to a particular machine learning domain.

### What area(s) of MLflow is this feature request about?

- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow model serving, deployment tools, Spark UDFs
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/prompt`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/projects`: MLproject format, project running backends
- [x] `area/uiux`: Front-end, user experience, plotting
- [ ] `area/docs`: MLflow documentation pages

**Comment by LockedThread** [comment]
Hi maintainers — I'm ready to implement this as a small frontend-only change to `ExperimentViewRunsColumnSelector` (CSS `resize: both` on desktop, disabled on `xs`, no persistence, no API/state changes).

Could someone please triage this issue and apply the `ready` label when appropriate? I'll open the PR once that label is present so it isn't auto-closed by the contribution policy.

**Comment by LockedThread** [comment]
I am going to begin using my fork in the mean time because this is very annoying for my developers. Figured I should still make the PR (in draft mode) for the sake of transparency.
